docker-compose.yml

```
@@ -0,0 +1,83 @@
+version: '3.8'
+
+services:
+  db:
+    image: 'mysql:9.0.1'
+    restart: always
+    working_dir: /app
+    environment:
+      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
+      MYSQL_DATABASE: ${MYSQL_DATABASE}
+      MYSQL_USER: ${MYSQL_USER}
+      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
+    volumes:
+      - db_data:/var/lib/mysql
+    networks:
+      - internal
+  api:
+    build:
+      context: ./backend
+      dockerfile: Dockerfile
+    command: >
+      sh -c "sleep 30 && npm run prisma-up && npm run prisma-gen && npm run start"
+    depends_on:
+      - db
+    environment:
+      DATABASE_URL: "mysql://root:${MYSQL_ROOT_PASSWORD}@db:3306/${MYSQL_DATABASE}"
+      NODE_ENV: production
+      HEX_USER_ID_OFFSET: 0
+      PORT: 3000
+      SMTP_PORT: 465
+      SMTP_HOST: ${SMTP_HOST}
+      SMTP_USERNAME: ${SMTP_USERNAME}
+      SMTP_PASSWORD: ${SMTP_PASSWORD}
+      SMTP_SENDER: ${SMTP_SENDER}
+      JWT_SECRET: ${JWT_SECRET}
+      RECAPTCHA_SECRET_KEY: ${RECAPTCHA_SECRET_KEY}
+      SIGN_IN_URL: ${PUBLIC_FRONTEND_URL}/auth/login
+      RESET_PASSWORD_URL: ${PUBLIC_FRONTEND_URL}/auth/password
+      CORS_ORIGIN: ${PUBLIC_FRONTEND_URL}
+      GOOGLE_AUTH_CLIENT_ID: none
+    networks:
+      - caddy
+      - internal
+    labels:
+      caddy: ${BACKEND_FQDN}
+      caddy.reverse_proxy: "{{upstreams http 3000}}"
+
+  web:
+    build:
+      context: ./frontend
+      dockerfile: Dockerfile
+    env_file:
+      - frontend/.env
```

Found a workaround for the requirement of using one entire .env for the docker-based deployment:

```suggestion
    build:
      context: .
      dockerfile: ./frontend/Dockerfile
    env_file:
      - .env
```